internal/build.go

```diff
@@ -75,3 +82,31 @@ func buildCommandLoaderFilter(bundleMode bool, ignore []string) func(string, os.
 		return util.LoaderFilter{Ignore: ignore}.Apply(abspath, info, depth)
 	}
 }
+
+func assertUniqueRuleIds(paths []string) error {
+	rules, err := util.RetrieveRules(paths)
+	if err != nil {
+		return err
+	}
+
+	pathsByVisitedRuleIds := make(map[string]string)
+
+	for _, rule := range rules {
+		rule.PublicId = strings.ToUpper(rule.PublicId)
```

Comment:
Just a note that we might not want to make this assertion case insensitive. Not sure yet though, depends on the discussion on Slack.

This `publicId` ends up in a database in a column, which in PostgreSQL is case sensitive: https://stackoverflow.com/a/20880247

Reply:
For the time being, I've changed the assertion to be case-sensitive. Based on the outcome of this [slack discussion](https://snyk.slack.com/archives/C022H54L8PJ/p1640857469331600), it may be changed back to be case-insensitive.